llvm/include/llvm/Passes/PassPlugin.h

```diff
@@ -107,7 +107,7 @@ class PassPlugin {
 ///   };
 /// }
 /// ```
-extern "C" ::llvm::PassPluginLibraryInfo LLVM_ATTRIBUTE_WEAK
+extern "C" ::llvm::PassPluginLibraryInfo LLVM_ATTRIBUTE_WEAK LLVM_ABI_EXPORT
```

Comment:
I find this a little misleading ... I think I would personally prefer:

```c
#if defined(_WIN32)
__declspec(dllexport)
#endif
```

The reasoning is that this is not the LLVM ABI but the plugin's ABI. But, I think that getting things working this way with a reference to an issue to clean up is fine.

How does this work within LLVM however? The declaration should be:

```c
extern "C" ::llvm::PassPluginLibraryInfo __declspec(dllimport)
llvmGetPassPluginInfo();
```

when this is being referenced in LLVM itself.

Reply:
I could change it to that or maybe make a new macro for plugins with a TODO note, I guess i could also remove LLVM_ATTRIBUTE_WEAK for windows builds as well.

> How does this work within LLVM however? The declaration should be:

I don't think it would ever be used in a dllimport mode since there can be multiple plugins.

Reply:
Because it is dynamically bound, yes, it cannot be `__declspec(dllimport)`. This either has to be passed indirectly or be looked up dynamically with `GetProcAddress`.